modules/opensearch/opensearch.module

```diff
@@ -154,10 +154,234 @@ function opensearch_admin_settings($form_state) {
     '#default_value' => variable_get('ting_search_profile', ''),
   );
 
+  /**
+   * Autocomplete settings.
+   */
+  $form['opensearch_autocomplete'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Opensearch autocomplete settings'),
+    '#tree' => FALSE,
+  );
+
+  $default_settings = opensearch_search_autocomplete_settings();
+
+  // Attach the CSS to the form
+  $form['opensearch_autocomplete']['#attached'] = array(
+    'css' => array(
+      'type' => 'file',
+      'data' => drupal_get_path('module', 'opensearch') . '/css/opensearch_autocomplete_admin_form.css',
+    ),
+  );
+
+  $form['opensearch_autocomplete']['opensearch_search_autocomplete_webservice'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Autocomplete URL', array(), array('context' => 'opensearch_search_autocomplete')),
+    '#tree' => FALSE,
+    '#collapsible' => TRUE,
+  );
+
+  $form['opensearch_autocomplete']['opensearch_search_autocomplete_webservice']['opensearch_search_autocomplete_suggestion_url'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Autocomplete URL', array(), array('context' => 'opensearch_search_autocomplete')),
+    '#size' => 73,
+    '#description' => t('URL to Autocomplete webservice. F.ex.: http://opensuggestion.addi.dk/1.0/', array(), array('context' => 'opensearch_search_autocomplete')),
+    '#default_value' => variable_get('opensearch_search_autocomplete_suggestion_url',
+      NULL),
+    '#required' => TRUE,
+    '#element_validate' => array('opensearch_search_autocomplete_admin_validate'),
+  );
+
+  $form['opensearch_autocomplete']['opensearch_search_autocomplete_webservice']['opensearch_search_autocomplete_method'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Autocomplete method', array(), array('context' => 'opensearch_search_autocomplete')),
+    '#size' => 73,
+    '#description' => t('The method to use. Currently there are two methods: terms and facets. Version 2.0 exposes facetSpell method', array(), array('context' => 'opensearch_search_autocomplete')),
+    '#default_value' => variable_get('opensearch_search_autocomplete_method',
+      'facets'),
+    '#required' => TRUE,
+  );
+
+  $form['opensearch_autocomplete']['opensearch_search_autocomplete_settings'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Autocomplete settings', array(), array('context' => 'opensearch_search_autocomplete')),
+    '#tree' => TRUE,
+    '#collapsible' => TRUE,
```

Comment:
This will flood the form with *lot* of settings. This can be overwhelming - especially since `ding_provider` will expose th the form during the installer.

I have two suggestions about how to handle this:

1. The easiest way would probably be to collapse the form by default.
2. Another approach could be to add a tab to the `opensearch` provider form and expose settings which have a default value there.

Reply:
I've updated the broken CSS. 
I think it's fine now after the applied CCS.
What do you think?

Reply:
:+1: